services/catalog/src/simcore_service_catalog/db/repositories/services.py

```diff
@@ -1,55 +1,125 @@
 import logging
-from typing import List, Optional
+from typing import List, Optional, Tuple
 
 import sqlalchemy as sa
+from aiopg.sa import SAConnection
 from aiopg.sa.result import RowProxy
 from models_library.services import ServiceAccessRightsAtDB, ServiceMetaDataAtDB
 from psycopg2.errors import ForeignKeyViolation  # pylint: disable=no-name-in-module
 from sqlalchemy import literal_column
 from sqlalchemy.dialects.postgresql import insert
 from sqlalchemy.sql import and_, or_
+from sqlalchemy.sql.selectable import Select
 
-from ..tables import services_access_rights, services_meta_data
+from ..tables import services_access_rights, services_meta_data, users
 from ._base import BaseRepository
 
 logger = logging.getLogger(__name__)
 
 
+def _make_list_services_query(
+    gids: Optional[List[int]] = None,
+    execute_access: Optional[bool] = None,
+    write_access: Optional[bool] = None,
+    combine_access_with_and: Optional[bool] = True,
+    product_name: Optional[str] = None,
+) -> Select:
+    query = sa.select([services_meta_data])
+    if gids or execute_access or write_access:
+        access_query_part = and_(
+            services_access_rights.c.execute_access if execute_access else True,
+            services_access_rights.c.write_access if write_access else True,
+        )
+        if not combine_access_with_and:
```

Comment:
Are you sure about having the __same defaults__ for both ``and_`` and ``or_``? Take into account that ``A & False == False`` and ``A || True == True`` regardless of ``A``

Shouldn't be something like:

```python
       
         logic_operator = and_ if combine_access_with_and else or_
         default = True if combine_access_with_and else False
         access_query_part = logic_operator(
            services_access_rights.c.execute_access if execute_access else default,
            services_access_rights.c.write_access if write_access else default,
        )
```

Reply:
yes you are right. the defaults do not make sense. but since I'm not aiming at pushing this at the moment...

Reply:
done